frontend/src/src/PrivateRoute.js

```diff
@@ -1,13 +1,22 @@
 import React from 'react';
-import { Route, Redirect } from 'react-router-dom';
+import {Route, Redirect, useLocation} from 'react-router-dom';
```

Comment:
this is showing up in a lot of places, it looks like your autoformatter is set to "no spaces" on named imports--if it's not too hard, would be nice to change that so you're not battling against others when you make changes. I know I've had cases where prettier is haunted, so if its giving you trouble, it's easy enough to figure it out another time and submit a formatting PR that minimal review